src/component-pipeline-stack.ts

```diff
@@ -75,12 +75,24 @@ export interface PipelineConfig {
    */
   readonly codeArtifactRepositoryArn?: string;
   /**
-   *
    *
    * @readonly
    * @type {?string}
    */
   readonly notificationTopicArn?: string;
+
+  /**
+   * The environment variables that your builds can use.
+   * @readonly
+   */
+  readonly environment?: NonNullable<CodeBuildOptions['buildEnvironment']>['environmentVariables'];
+
+  /**
+   *
+   * @readonly
+   * @type {?string}
+   */
+  readonly roles?: CodeBuildOptions['rolePolicy'];
```

Comment:
This prop deserves some documentation. I would probably also rename it to `rolePolicy`. In fact, why not move up a layer to just `assetPublishingCodeBuildDefaults: CodeBuildOptions` and give room to set anything you want.